EECS3311-PD1/src/main/java/UI/Login/LoginView.java

```diff
@@ -0,0 +1,100 @@
+package UI.Login;
+
+import javafx.event.ActionEvent;
+import javafx.event.Event;
+import javafx.event.EventHandler;
+import javafx.event.EventType;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Button;
+import javafx.scene.control.PasswordField;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.VBox;
+import org.example.userDb.*;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class LoginView extends VBox {
+    static public class LoginSuccessfulEvent extends Event {
+        public LoginSuccessfulEvent(EventType<LoginSuccessfulEvent> eventType) {
+            super(eventType);
+        }
+    }
+
+    private EventHandler<LoginSuccessfulEvent> onLoginSuccessful;
+
+    UserDbModel model;
+
+    public TextField usernameTF;
+    public PasswordField passwordTF;
+    public Button loginButton;
+
+    public LoginView() {
+        model = new UserDbModel(
+                new UserSingleTableDatabase(
+                        new LocalJsonTableDataSourceFactory(
+                                Objects.requireNonNull(
+                                        getClass().getResource("/userdb.json")).getPath()
+                        )
+                )
+        );
+
+        onLoginSuccessful = null;
+
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource(
+                "/fxml/LoginView.fxml"));
+        fxmlLoader.setRoot(this);
+        fxmlLoader.setController(this);
+
+        try {
+            fxmlLoader.load();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+
```

Comment:
New to the double colon operator but from what I read online you are calling the onLoginClick, onUsenameChange, and onPasswordChange methods in the LoginView class given that "this" is of type LoginView.